Assets/MixedRealityToolkit.SDK/Experimental/Features/Dwell/DwellHandler.cs

```diff
@@ -0,0 +1,243 @@
+
+        [Header("Dwell Events")]
+        [SerializeField]
+        private PointerUnityEvent DwellIntended = new PointerUnityEvent();
+
+        [SerializeField]
+        private PointerUnityEvent DwellStarted = new PointerUnityEvent();
+
+        [SerializeField]
+        private PointerUnityEvent DwellCompleted = new PointerUnityEvent();
+
+        [SerializeField]
+        private PointerUnityEvent DwellCanceled = new PointerUnityEvent();
+
+        /// <summary>
+        /// Invoke this event if the dwell is interrupted with alternate modality
+        /// </summary>
+        public UnityAction CancelDwellAction;
+
+        /// <summary>
+        /// Property exposing the computation for what percentage of dwell has progressed.
+        /// </summary>
+        public virtual float DwellProgress
+        {
+            get
+            {
+                switch (CurrentDwellState)
+                {
+                    case DwellStateType.None:
+                    case DwellStateType.FocusGained:
+                        return 0;
+                    case DwellStateType.DwellStarted:
+                        return Mathf.Clamp((float)(dwellProfile.TimeToCompleteDwell - FillTimer) / dwellProfile.TimeToCompleteDwell, 0f, 1f);
+                    case DwellStateType.DwellCompleted:
+                        return 1;
+                    case DwellStateType.DwellCanceled:
+                        if ((DateTime.UtcNow - focusExitTime).TotalSeconds > dwellProfile.TimeToAllowDwellResume)
+                        {
+                            CurrentDwellState = DwellStateType.None;
+                            return 0;
+                        }
+                        else if (dwellProfile.TimeToAllowDwellResume > 0)
+                        {
+                            return Mathf.Clamp((float)(dwellProfile.TimeToCompleteDwell - FillTimer) / dwellProfile.TimeToCompleteDwell, 0f, 1f);
+                        }
+                        break;
+                    default:
+                        return 0;
+                }
+
+                return 0;
+            }
+        }
+
+        /// <summary>
+        /// Cached pointer reference to track focus events maps to teh same pointer that initiated dwell
+        /// </summary>
+        private IMixedRealityPointer pointer;
+
+        private DateTime focusEnterTime = DateTime.MaxValue;
+        private DateTime focusExitTime = DateTime.MaxValue;
+
+        /// <summary>
+        /// Exposes whether the target has focus from the pointer defined in dwell profile settings
+        /// </summary>
+        protected bool HasFocus { get; private set; }
+
+        /// <summary>
+        /// Captures the dwell status 
+        /// </summary>
+        protected DwellStateType CurrentDwellState = DwellStateType.None;
+
+        /// <summary>
+        /// Abstracted value for the how long the dwelled object still needs to be focused to comlete the dwell action
```

Comment:
>long [](start = 41, length = 4)

Nit: What are the units on this, or is it unitless? (If unitless, documenting that it is unitless is helpful)